manual/src/refman/expr.etex

```diff
@@ -570,6 +786,13 @@ selected. If none of the patterns matches the value of @expr@, the
 exception value is raised again, thereby transparently ``passing
 through'' the @'try'@ construct.
 
+\begin{caml_example}{toplevel}
+let option_exception f x =
+  try Some (f x) with
+  | Assert_failure _ as e -> raise e
+  | _ -> None;;
+\end{caml_example}
```

Comment:
Looks like you're reusing an example from the "Patterns" section.  This example does not illustrate the automatic "reraise"...

Reply:
So if I changed it to:

`Assert_failure (a, b, c) -> raise (Assert_failure (a, b, c))`

It would be detected as a reraise?

Reply:
Sorry if I was unclear.  The "automatic reraise" behavior I'm alluding to is the fact that
```
  try e1 with Not_found -> e2
```
is equivalent to
```
  try e1 with Not_found -> e2 | exn -> raise exn
```
Surely you want to illustrate this fact?

Moreover, exception handlers with catch-all cases that don't re-raise the exception are frowned upon.  In your example, if function `f` raises `Invalid_argument` or `Out_of_memory` or `Sys.break`, all these exceptions are caught and turned into a non-informative `None` result.  Don't do that, and don't use that as an example.